src/Form.php

```diff
@@ -5,18 +5,22 @@
 namespace Yiisoft\Form;
 
 use Yiisoft\Strings\Inflector;
+use Yiisoft\Validator\DataSetInterface;
+use Yiisoft\Validator\Validator;
 
-abstract class Form implements FormInterface
+abstract class Form implements FormInterface, DataSetInterface
 {
     private array $attributes;
     private array $attributesLabels;
     private array $attributesErrors = [];
+    private array $attributesRules = [];
     private ?Inflector $inflector = null;
 
     public function __construct()
     {
         $this->attributes = $this->attributes();
         $this->attributesLabels = $this->attributeLabels();
+        $this->attributesRules = $this->rules();
```

Comment:
I think you need to keep empty `rules` into this class, because form can exist without any rules

Reply:
Done.